app/src/main/java/com/duckduckgo/app/browser/downloader/FileDownloader.kt

```diff
@@ -57,8 +58,8 @@ class FileDownloader @Inject constructor(
     }
 }
 
-fun FileDownloader.PendingFileDownload.guessFileName(): String {
-    val guessedFileName = URLUtil.guessFileName(url, contentDisposition, mimeType)
+fun FileDownloader.PendingFileDownload.guessFileName(filenameExtractor: FilenameExtractor): String {
```

Comment:
No strong opinion, and not really important, just talking out loud.

It feels a bit weird that `PendingFileDownload` has a method `guessFileName` that receives a `FileNameExtractor` to guess the file name. I would expect the other way around: `FileNameExtractor.guessFileNameFrom(pendingFileDownload)` as a extension function that receives a `PendingFileDownload`. Inside that method we map `PendingFileDownload` to `(url, contentDisposition, mimeType)`

Reply:
Agree, this is weird. I've removed that extension function on `pendingFileDownload` and put the logic all in `FileNameExtractor`